pkg/operation/botanist/component/logging/eventlogger/event_logger_test.go

```diff
@@ -0,0 +1,565 @@
+		managedResourceSecret      *corev1.Secret
+		eventLoggerServiceAccount  *corev1.ServiceAccount
+		seedEventLoggerRole        *rbacv1.Role
+		seedEventLoggerRoleBinding *rbacv1.RoleBinding
+		eventLoggerDeployment      *appsv1.Deployment
+		vpa                        *vpaautoscalingv1.VerticalPodAutoscaler
+
+		eventLoggerDeployer component.Deployer
+		fakeSecretManager   secretsmanager.Interface
+
+		clusterRoleForShoot = func() *rbacv1.ClusterRole {
+			return &rbacv1.ClusterRole{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "rbac.authorization.k8s.io/v1",
+					Kind:       "ClusterRole",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: name,
+					Labels: map[string]string{
+						v1beta1constants.LabelApp:   name,
+						v1beta1constants.LabelRole:  v1beta1constants.LabelLogging,
+						v1beta1constants.GardenRole: v1beta1constants.GardenRoleLogging,
+					},
+				},
+				Rules: []rbacv1.PolicyRule{
+					{
+						APIGroups: []string{
+							"",
+						},
+						Resources: []string{
+							"events",
+						},
+						Verbs: []string{
+							"get",
+							"list",
+							"watch",
+						},
+					},
+				},
+			}
+		}
+
+		roleBindingFor = func(clusterType component.ClusterType, namespace string, removeResourceVersion bool) *rbacv1.RoleBinding {
+			obj := &rbacv1.RoleBinding{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: rbacv1.SchemeGroupVersion.String(),
+					Kind:       "RoleBinding",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "gardener.cloud:logging:event-logger",
+					Namespace: namespace,
+					Labels: map[string]string{
+						v1beta1constants.LabelApp:   name,
+						v1beta1constants.LabelRole:  v1beta1constants.LabelLogging,
+						v1beta1constants.GardenRole: v1beta1constants.GardenRoleLogging,
+					},
+					ResourceVersion: "1",
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: rbacv1.GroupName,
+					Kind:     "ClusterRole",
+					Name:     name,
+				},
+				Subjects: []rbacv1.Subject{{
+					Kind:      rbacv1.ServiceAccountKind,
+					Name:      name,
+					Namespace: metav1.NamespaceSystem,
+				}},
+			}
+
+			if clusterType == component.ClusterTypeSeed {
+				obj.Subjects[0].Namespace = namespace
+				obj.Name = name
+				obj.RoleRef.Kind = "Role"
+			}
+
+			if removeResourceVersion {
+				obj.ResourceVersion = ""
+			}
+
+			return obj
+		}
+
+		vpaUpdateMode    = vpaautoscalingv1.UpdateModeAuto
+		controlledValues = vpaautoscalingv1.ContainerControlledValuesRequestsOnly
+	)
+
+	type newEventLoggerArgs struct {
+		client         client.Client
+		namespace      string
+		secretsManager manager.Interface
+		image          string
+		err            error
+	}
+
+	DescribeTable("#New",
+		func(args newEventLoggerArgs) {
+			deployer, err := New(
+				args.client,
+				args.namespace,
+				args.secretsManager,
+				Values{
+					Image:    args.image,
+					Replicas: 1,
+				},
+			)
+			if args.err != nil {
+				Expect(err).To(Equal(args.err))
```

Comment:
You can pass a `types.GomegaMatcher` to the func and avoid the if statement. It would look something like this:

```
func (args newEventLoggerArgs, matchError, matchDeployer types.GomegaMatcher) {
    ....
    Expect(err).To(matchError)
    Expect(deployer).To(matchDeployer)
}
Entry("pass args with nil shoot and client", newEventLoggerArgs {
    ....
    },
    MatchError(ContainsSubstring("client cannot be nil")),
    BeNil(),
}
```

Also, notice that the error matcher is `MatchError(ContainsSubstring("client cannot be nil")),` which is more flexible than trying to check for `Equal` error objects.